Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-view/test-view.component.ts

```diff
@@ -1,19 +1,174 @@
-﻿import { Component } from '@angular/core';
+﻿import { Component, OnInit } from '@angular/core';
 import { MdDialog } from '@angular/material';
 import { TestLaunchDialogComponent } from '../test-settings/test-launch-dialog.component';
 import { DeleteTestDialogComponent } from '../tests-dashboard/delete-test-dialog.component';
+import { Question } from '../../questions/question.model';
+import { Category } from '../../questions/category.model';
+import { TestService } from '../tests.service';
+import { Router, ActivatedRoute } from '@angular/router';
+import { MdSnackBar, MdSnackBarConfig } from '@angular/material';
+import { DifficultyLevel } from '../../questions/enum-difficultylevel';
+import { Test } from '../tests.model';
+import { QuestionBase } from '../../questions/question';
+import { QuestionType } from '../../questions/enum-questiontype';
+import { TestAttendees } from '../../conduct/register/register.model';
+import { DuplicateTestDialogComponent } from '../tests-dashboard/duplicate-test-dialog.component';
 
 @Component({
     moduleId: module.id,
     selector: 'test-view',
     templateUrl: 'test-view.html'
 })
 
-export class TestViewComponent {
-    constructor(public dialog: MdDialog) { }
+export class TestViewComponent implements OnInit {
+    testDetails: Test;
+    testId: number;
+    totalNumberOfQuestions: number[] = [];
+    QuestionType = QuestionType;
+    DifficultyLevel = DifficultyLevel;
+    optionName: string[] = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j'];
+    isDeleteAllowed: boolean;
+    Tests: Test[] = new Array<Test>();
```

Comment:
initialization should go in a constructor and variable name should be camelCase.